Usage documentation for extensions

jkarsrud opened the pull request:
Changed the documentation of the extensions to be more usage docs than developer docs. Also added a page to /developers with extension-developing guidelines and the roadmap.

jkarsrud committed:
extensions

jkarsrud committed:
Merge remote-tracking branch 'upstream/master'

jkarsrud committed:
Added extensions to the docs

jkarsrud committed:
fixed line endings

jkarsrud committed:
Merge remote-tracking branch 'upstream/master'

jkarsrud committed:
Merge remote-tracking branch 'upstream/master'

jkarsrud committed:
Added more usage-centric docs on extensions

jkarsrud committed:
Removed undocumented usage docs

jkarsrud committed:
Changed from warning to note

jkarsrud commented:
Hmm, this got a few more commits than I wanted, but the end result should be good :)

jodal commented on developers/extensions.rst:
```diff
@@ -0,0 +1,121 @@
+.. extensions:
```
You need a "_" prefix for "extensions" to be ref-able.

jodal commented on developers/extensions.rst:
```diff
@@ -0,0 +1,121 @@
+
+===================
+Building Extensions
+===================
```
"====" over and under the header is a level higher in the header hierarchy than just below, so you can use "====" below the next headers, and not need to use "^^^^" on the lowest levels. This works nicely as is, just thought to let you know how it works.

jodal commented on developers/extensions.rst:
```diff
@@ -0,0 +1,121 @@
+expected to increase in the near future.
+
+Create: ``var instance = ext.create([options]);``
+-----------------------------------------
```
Guess you'll get a warning here about header and line not being equally long.

jodal commented on extensions/buster-amd.rst:
```diff
@@ -1,7 +1,66 @@
+Install
+-------
+
+Installation is done using npm: ``npm install buster-amd``.
```
Would be nice to make the install command a code block, just to make it more visible.

jodal commented on extensions/buster-amd.rst:
```diff
@@ -1,7 +1,66 @@
+is an optional function that translate Buster.JS paths (which are absolute,
+e.g. ``/test/my-test.js``) to AMD friendly module IDs. The default
+mapper converts ``/test/my-test.js`` to ``test/my-test``, i.e. strips leading
+slash and file suffix. Use this option to e.g. use AMD loader plugins.::
```
Will ".::" on the end of  the line be rendered as "." or ".:"? If a colon doesn't naturally fit on the end of the sentence, I would put the "::" on a line of its own.

jkarsrud replied:
It seems it does render ".:", so I put "::" on a separate line.

jodal commented on extensions/buster-jstestdriver.rst:
```diff
@@ -4,4 +4,29 @@
+
+Usage
+-----
+Load in your configuration file::
```
Missing empty line after the header.

jodal commented on extensions/buster-lint.rst:
```diff
@@ -1,7 +1,74 @@
+
+Installation is done using npm: ``npm install buster-lint``.
+
+Usage
```
Should be a header.

jodal commented on roadmap.rst:
```diff
@@ -0,0 +1,120 @@
+
+A revamped ``buster-static`` that can do both Node and Browser tests
+from within the browser is included in the default installation. Installers
+for OSX and Windows.
```
s/Node/Node.js/, s/Browser/browser/, s/OSX/OS X/

jkarsrud committed:
Fixes some small issues with headers and code blocks

jkarsrud committed:
Fixed some small typoss

jkarsrud commented:
Fixed the issues now, cleaned up the headers as well. Using ==== and ----- does look a lot cleaner. Let me know if there's anything else preventing this from being pulled :)

jkarsrud committed:
updated roadmap with buster-ci section

jodal commented:
Looks good :-)
